### ISSUE TITLE
`MissingSummary` triggers `IllegalArgumentException` 

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#1444]
Consider the following code:
```java
interface Dummy {
  /**
   * @return an integer
   * @see <a href="https://example.com">Example</a>
   */
  int myMethod();
}
```

Compile the code as follows (Caffeine must be added to the classpath explicitly due to #1398):
```sh
javac \
  -XDcompilePolicy=simple \
  -processorpath error_prone_core-2.3.4-with-dependencies.jar:caffeine-2.8.0.jar \
  '-Xplugin:ErrorProne -XepDisableAllChecks -Xep:MissingSummary:WARN' \
  Dummy.java
```

This causes the following failure:
```
Dummy.java:6: error: An unhandled exception was thrown by the Error Prone static analysis plugin.
  int myMethod();
      ^
     Please report this at https://github.com/google/error-prone/issues/new and include the following:
  
     error-prone version: 2.3.4
     BugPattern: MissingSummary
     Stack Trace:
     java.lang.IllegalArgumentException: Start [53] should not be after end [-1]
  	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:300)
  	at com.google.errorprone.fixes.IndexedPosition.<init>(IndexedPosition.java:33)
  	at com.google.errorprone.fixes.SuggestedFix$Builder.replace(SuggestedFix.java:204)
  	at com.google.errorprone.fixes.SuggestedFix.replace(SuggestedFix.java:110)
  	at com.google.errorprone.bugpatterns.javadoc.Utils.replace(Utils.java:68)
  	at com.google.errorprone.bugpatterns.javadoc.MissingSummary.generateSeeFix(MissingSummary.java:154)
  	at com.google.errorprone.bugpatterns.javadoc.MissingSummary.handle(MissingSummary.java:117)
  	at com.google.errorprone.bugpatterns.javadoc.MissingSummary.matchMethod(MissingSummary.java:84)
  	at com.google.errorprone.scanner.ErrorProneScanner.processMatchers(ErrorProneScanner.java:451)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitMethod(ErrorProneScanner.java:741)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitMethod(ErrorProneScanner.java:152)
  	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCMethodDecl.accept(JCTree.java:898)
  	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:72)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:46)
  	at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
  	at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
  	at jdk.compiler/com.sun.source.util.TreeScanner.visitClass(TreeScanner.java:187)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:550)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:152)
  	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:808)
  	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:72)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:46)
  	at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
  	at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
  	at jdk.compiler/com.sun.source.util.TreeScanner.visitCompilationUnit(TreeScanner.java:144)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:562)
  	at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:152)
  	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:591)
  	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
  	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:56)
  	at com.google.errorprone.scanner.ErrorProneScannerTransformer.apply(ErrorProneScannerTransformer.java:43)
  	at com.google.errorprone.ErrorProneAnalyzer.finished(ErrorProneAnalyzer.java:152)
  	at jdk.compiler/com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:132)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1418)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1365)
  	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:960)
  	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:311)
  	at jdk.compiler/com.sun.tools.javac.main.Main.compile(Main.java:170)
  	at jdk.compiler/com.sun.tools.javac.Main.compile(Main.java:57)
  	at jdk.compiler/com.sun.tools.javac.Main.main(Main.java:43)
1 error
```

If the `@see` contains a "regular" reference instead of an HTML link then the exception is not thrown.

**Comment by graememorgan** [comment]
Thanks for the report. I've sent out a fix internally.

There seems to be some issue with javac's parsing of Javadocs. It doesn't output an end position for @see in this case.

**Comment by Stephan202** [comment]
This issue was resolved by 39d25066291d81d162fca62e3ad4d106a94545d0; verified fixed against `master` (65e5dff6ddbefeab0552a1f60ded55a34fc76f51).
